# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/gtk/mozwayland/mozwayland.h
# Commit: 6a13d76d61b8
# Full Hash: 6a13d76d61b876a3e4f22a42fecb27de5c106121
# Author: stransky <stransky@redhat.com>
# Date: 2024-09-13 04:28:43
# Description:
#   Bug 1917558 [Wayland] Explicitly request wl-pointer v.3 interface r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D221617
# ==============================================================================

diff -r a8e3d92b384c -r 6a13d76d61b8 widget/gtk/mozwayland/mozwayland.h
--- a/widget/gtk/mozwayland/mozwayland.h	Thu Sep 12 11:49:58 2024 +0000
+++ b/widget/gtk/mozwayland/mozwayland.h	Thu Sep 12 12:15:42 2024 +0000
@@ -43,6 +43,10 @@
 #  define WL_MARSHAL_FLAG_DESTROY (1 << 0)
 #endif
 
+#ifndef WL_SURFACE_DAMAGE_BUFFER_SINCE_VERSION
+#  define WL_SURFACE_DAMAGE_BUFFER_SINCE_VERSION 4
+#endif
+
 /* We need implement some missing functions from wayland-client-protocol.h
  */
 #ifndef WL_DATA_DEVICE_MANAGER_DND_ACTION_ENUM
@@ -135,6 +139,10 @@
 }
 #endif
 
+#ifndef WL_POINTER_RELEASE_SINCE_VERSION
+#  define WL_POINTER_RELEASE_SINCE_VERSION 3
+#endif
+
 #ifndef WL_POINTER_AXIS_ENUM
 #  define WL_POINTER_AXIS_ENUM
 /**